ui/packages/catalog-ui-search/src/main/webapp/js/widgets/cesium.base.line.js

```diff
@@ -0,0 +1,182 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+
+const Marionette = require('marionette');
+const Cesium = require('cesium');
+const _ = require('underscore');
+const wreqr = require('wreqr');
+const NotificationView = require('./notification.view');
+const DrawingController = require('./drawing.controller');
+
+"use strict";
+
+const CAMERA_MAGNITUDE_THRESHOLD = 8000000;
+
+const capitalize = (value) => value.charAt(0).toUpperCase() + value.slice(1);
+
+const getCurrentMagnitudeFromView = (view) => {
+    return view.map.camera.getMagnitude();
+}
+
+const needsRedraw = (view) => {
+    const currentMagnitude = getCurrentMagnitudeFromView(view);
+
+    if (view.cameraMagnitude < CAMERA_MAGNITUDE_THRESHOLD && currentMagnitude > CAMERA_MAGNITUDE_THRESHOLD) {
+        return true;
+    }
+    if (view.cameraMagnitude > CAMERA_MAGNITUDE_THRESHOLD && currentMagnitude < CAMERA_MAGNITUDE_THRESHOLD) {
+        return true;
+    }
+
+    return false;
+}
+
+class GeometryRenderView extends Marionette.View {
+  constructor(options) {
+    super(...options);
+
+    Object.assign(this, {...options});
+  }
+
+  updatePrimitive = () => {
+    this.drawGeometry(this.model);
+  }
+
+  drawGeometry = (model) => {
+    throw new Error('This method must be overwritten by implementing geometry!');
+  }
+
+  constructLinePrimitive = (coordinates) => {
+    const color = this.model.get('color');
+
+    return {
+      width: 8,
+      material: Cesium.Material.fromType('PolylineOutline', {
+          color: color ? Cesium.Color.fromCssColorString(color) : Cesium.Color.KHAKI,
+          outlineColor: Cesium.Color.WHITE,
+          outlineWidth: 4
+      }),
+      id: 'userDrawing',
+      positions: Cesium.Cartesian3.fromDegreesArray(_.flatten(coordinates))
+    };
+  }
+
+  listenForCameraChange = () => {
+    this.map.scene.camera.moveStart.addEventListener(this.handleCameraMoveStart, this);
+    this.map.scene.camera.moveEnd.addEventListener(this.handleCameraMoveEnd, this);
+  }
+
+  handleCameraMoveStart = () => {
+    this.animationFrameId = window.requestAnimationFrame(function() {
```

Comment:
✏️  Could be an arrow function / lambda

Reply:
True, but since fat arrow functions don't support current scope binding I would need to capture `this` reference so I opted with keeping `function(){}.bind(this)`